### PR TITLE
fix: align RSI with Wilder EMA

### LINKED ISSUE #1341: improvement: RSI 计算算法与券商/TradingView 不一致（SMA vs Wilder's EMA）
## 问题描述

在做Issue #1202 P5时观察到，如果未来有用户反馈"DSA 的 RSI 和我券商看到的不一样"。

那么原因如下：当前仓库中 RSI 的实现（`src/stock_analyzer.py` 和 `src/services/alert_indicators.py`）使用的是 **SMA（简单移动平均）** 方式计算平均涨幅和平均跌幅：

```python
avg_gain = gain.rolling(window=period).mean()
avg_loss = loss.rolling(window=period).mean()
```

而业界标准（TradingView、通达信、同花顺、MetaStock 等）使用的是 **Wilder's EMA（指数移动平均 / SMMA）**：

```python
avg_gain = gain.ewm(alpha=1/period, adjust=False).mean()
avg_loss = loss.ewm(alpha=1/period, adjust=False).mean()
```

两者的区别在于：SMA 对窗口内所有样本等权，窗口外数据完全丢弃；Wilder's EMA 对历史数据按指数衰减加权，因此对同一组数据会产生不同的 RSI 数值。

## 影响范围

- `src/stock_analyzer.py`：分析报告中的 RSI_6 / RSI_12 / RSI_24
- `src/services/alert_indicators.py`：P5 告警规则中的 `rsi_threshold` 评估
- 用户如果基于券商看到的 RSI 值设置告警阈值（如 RSI > 70 超买告警），可能因为算法差异导致触发时机偏移

## 数值差异估计

- 对于常用周期 period=12-14、样本充足（>50 根 bar）的情况下，SMA RSI 与 EMA RSI 的稳态差异通常在 **2-5 个点** 以内。
- 只有在短周期（period=2-3）或价格波动剧烈时差异才会显著放大。

## 建议

1. 将 `stock_analyzer.py` 和 `alert_indicators.py` 中的 RSI 计算统一升级为 Wilder's EMA 实现。
2. 两处修改需同步进行，保持分析报告与告警触发的 RSI 数值一致性。
3. 修改后需要更新对应的 golden tests（`tests/test_alert_worker.py` 中的 RSI 相关断言）。
4. 可考虑在 `docs/alerts.md` 中标注 RSI 的具体计算公式，方便用户与外部工具交叉验证。

## 优先级

低。当前 SMA 实现在项目内部是自洽的（报告和告警使用同一算法），不影响核心功能。但对于习惯与 TradingView / 通达信交叉对比的用户，数值偏差会造成困惑。

## 关联

- Issue #1202 P5（技术指标告警规则）
- `src/stock_analyzer.py` 第 304-337 行
- `src/services/alert_indicators.py` 第 385-392 行

### PR DESCRIPTION
**Pull request by massif-01**
## PR Type

- [x] fix
- [ ] feat
- [ ] refactor
- [x] docs
- [ ] chore
- [x] test

## Background And Problem

Fixes #1341

当前 `src/stock_analyzer.py` 与 `src/services/alert_indicators.py` 的 RSI 计算使用 rolling SMA 计算平均涨跌幅，和 Issue #1341 中指定的 Wilder's EMA / SMMA 口径不一致。用户对照 TradingView、通达信或券商软件设置 `rsi_threshold` 时，报告和告警里的 RSI 数值可能与外部口径产生偏差。

本 PR 将分析报告与告警规则两条路径统一调整为 pandas `ewm(alpha=1/period, adjust=False)` 口径。

## Scope Of Change

- `src/stock_analyzer.py`：报告侧 `RSI_6 / RSI_12 / RSI_24` 改用 Wilder's EMA / SMMA 口径，并同步函数说明。
- `src/services/alert_indicators.py`：告警侧 `rsi_threshold` 评估改用同一 RSI 口径。
- `tests/test_alert_worker.py`：更新 RSI golden，并增加旧 SMA 结果的负向参考断言。
- `tests/test_stock_analyzer_rsi.py`：新增报告侧 RSI golden 测试，并断言报告与告警路径在 `RSI_6 / RSI_12 / RSI_24` 周期上保持一致。
- `docs/alerts.md`：补充 P5 RSI 公式说明。
- `docs/CHANGELOG.md`：记录 `[Unreleased]` 改进条目。

未改动 API schema、数据库、Web UI、配置项、legacy JSON 或公共指标架构。

## Issue Link

Fixes #1341

## Verification Commands And Results

```bash
python -m pytest tests/test_alert_worker.py::AlertIndicatorHelperTestCase tests/test_stock_analyzer_rsi.py tests/test_alerts_docs.py
git diff --check HEAD~2..HEAD
./scripts/ci_gate.sh
```

关键输出/结论：

- `python -m pytest ...`：`21 passed`
- `git diff --check HEAD~2..HEAD`：无 whitespace 错误
- `./scripts/ci_gate.sh`：`2097 passed, 2 deselected, 76 warnings`，`backend-gate: all checks passed`

## Compatibility And Risk

这是用户可见的算法口径调整，不是行情数据异常修复：

- 分析报告中的 `RSI_6 / RSI_12 / RSI_24` 数值会变化。
- P5 告警规则 `rsi_threshold` 的观察值和触发时机可能变化。
- 本 PR 按 Issue #1341 明确指定的 pandas `ewm(alpha=1/period, adjust=False)` 口径调整 RSI。该实现与经典 Wilder SMA-seeded warm-up 或不同券商 / 图表软件在预热期可能存在细微差异；本 PR 不声明与所有外部工具 bit-level 完全一致。

没有运行时配置保存、清理、迁移或回填逻辑变更；已有配置不会被自动改写、清空或迁移。

## Rollback Plan

最小回滚方式：revert this PR。无需额外回滚配置、数据库或用户数据。

## EXTRACT_PROMPT Change (if applicable)

不涉及 `src/services/image_stock_extractor.py` 的 `EXTRACT_PROMPT`。

## Checklist

- [x] 本 PR 有明确动机和业务价值 / This PR has a clear motivation and value
- [x] 已提供可复现的验证命令与结果 / Reproducible verification commands and results are included
- [x] 已评估兼容性与风险 / Compatibility and risk have been assessed
- [x] 已提供回滚方案 / A rollback plan is provided
- [x] 若涉及用户可见变更，已同步更新相关文档与 `docs/CHANGELOG.md`；`README.md` 仅在首页级信息变化时更新，细节优先写入 `docs/*.md` / If user-visible changes are included, relevant docs and `docs/CHANGELOG.md` are updated; `README.md` is updated only for homepage-level changes, with details kept in `docs/*.md`

### CODE REVIEW

**Comment by massif-01** [comment]
@codex review

**Review by ZhuLinsen — COMMENTED** [review]
**评审结论**
- **必要性**：通过 + RSI 口径不一致会直接影响报告数值和告警触发判断，修复有明确业务必要性。
- **是否有对应 issue**：有，Fixes #1341。
- **PR 类型**：fix + 同步包含 docs/test；核心是修正 RSI 计算口径。
- **description 完整性**：完整 + 已说明背景、范围、兼容风险、验证结果、回滚方式，并符合 AGENTS.md 对用户可见变更同步文档与 `docs/CHANGELOG.md` 的要求。
- **是否可直接合入**：可 + CI 当前成功，diff 覆盖报告侧与告警侧 RSI 口径、文档和回归测试；`mergeable_state=blocked` 在本仓库规则下不单独构成阻断。

---
*🤖 此回复由 OpenReview Bot 自动生成，仅供参考。如有疑问请 @维护者。*
